src/meta/model/src/table.rs

```diff
@@ -136,6 +164,7 @@ pub struct Model {
     pub cdc_table_id: Option<String>,
     pub vnode_count: i32,
     pub webhook_info: Option<WebhookSourceInfo>,
+    pub engine: Engine,
```

Comment:
I was thinking whether this would break compatibility (adding a non-null field).

Then I found that 
1. `risedev d` on main, and create a table
2. checkout this branch, `risedev d`.

Then I cannot start cluster 
```
thread 'main' panicked at /Users/xxchan/Projects/risingwave/src/common/common_service/src/observer_manager.rs:76:62:
called `Result::unwrap()` on an `Err` value: Rpc(GrpcStatus(TonicStatusWrapper { inner: Status { code: Internal, message: "invalid hummock context 1", metadata: MetadataMap { headers: {"content-type": "application/grpc", "risingwave-error-bin": "GQAAAAAAAABpbnZhbGlkIGh1bW1vY2sgY29udGV4dCAxAAEHAAAAAAAAAGh1bW1vY2sA", "date": "Fri, 29 Nov 2024 10:06:25 GMT", "content-length": "0"} }, source: Some(ServerError { error: invalid hummock context 1, service_name: Some("hummock"), extra: Extra { score: None } }) }, client_side_service_name: Some("meta") }))
```
The error is a bit strange.. Not sure what's the cause.

Reply:
Let me take a look at whether I can reproduce it.

Reply:
Making the newly added field `engine` optional can resolve this issue. I think it is indeed a backward compatibility issue.

Reply:
@yezizp2012 Do you have any idea why the error is so strange. 🤪

Reply:
BTW, another way to fix it is to add column with default value in migration script. Then we can make this field required. Both OK to me.

Reply:
BTW, just nitpicking: if it's `Option<Engine>`, why bother adding `Hummock` engine. 🤪

Reply:
To me optional is used for backward compatibility, but the engine without hummock seems a bit weird.

Reply:
1. define the column `engine` without `not_null()` and set it as optional in model.
2. define the column `engine` with `not_null()` and `default(xxx)`, set it as non-optional in model. i.e: https://github.com/risingwavelabs/risingwave/blob/e9e3b24442cc55755c826ec9264fff909e553ef7/src/meta/model/migration/src/m20240911_083152_variable_vnode_count.rs#L3-L7

IIUC These two ways will not cause any backward compatibility issues. Besides please make sure that the definitions in the migration and model are consistent.

Reply:
Got it. Right now engine is an optional field, so it is nullable.

Reply:
For me a non-nullable column with default value can be much clearer. There are redundant semantics between `None` and `Some(Hummock)`.

As mentioned by @yezizp2012, variable vnode count work introduces a required field but not cause any backwards compat issues.